### PR TITLE
Enable nullability in AutoCompleteStringCollection

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in AutoCompleteStringCollection.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7047)

### CODE REVIEW

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good!

File: src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs [hunk 1]
```diff
@@ -194,51 +193,75 @@ public void RemoveAt(int index)
...
+        object? IList.this[int index]
         {
             get
             {
                 return this[index];
             }
             set
             {
-                this[index] = (string)value;
+                if (value is string valueAsString)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:204]: Before the change if value were not a string, this was an exception, now we handle in gracefully. @dreddy-work - thoughts?

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:204]: while some of the other interface overrides have return value to check on, this one ( couple below) are silently skipping. Given these are public, and documentation isn't very clear on what is expected incase of non string insertions, Maybe we should keep the existing behavior here.

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:204]: I agree, silent skipping opens opportunities for misuse, I would prefer an in-your-face  CastException.

**Review by Tanya-Solyanik — DISMISSED** [review]
Please retain cast exceptions where applicable.

File: src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs (hunk 1 above)
Comment (kant2002) [reply src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:204]: Should this behavior be captured using test?

File: src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs [hunk 2]
```diff
@@ -14,8 +12,8 @@ namespace System.Windows.Forms
     /// </summary>
     public class AutoCompleteStringCollection : IList
     {
-        CollectionChangeEventHandler onCollectionChanged;
-        private readonly ArrayList data = new ArrayList();
+        CollectionChangeEventHandler? _onCollectionChanged;
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/AutoCompleteStringCollection.cs:15]: nit: please add explicit access modifier

**Comment by dreddy-work** [comment]
@gpetrou , can you address feedback here and i will help merge it?

**Comment by dreddy-work** [comment]
@gpetrou, can you take a relook at this and rebase?

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**